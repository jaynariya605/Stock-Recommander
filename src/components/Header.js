import React from "react";

export const Header = () =>(

    <header className="header">
      <div className="content-container">
        <div className="header__content">
          <div className="header__title">
            <h1>Stock Recommander</h1>
          </div>   
        </div>
      </div>
    </header>

)