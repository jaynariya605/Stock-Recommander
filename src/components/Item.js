
const Item = ({ data, recomand }) => {
    
    return (
        <div className="list-item">
            <h3 className="list-item__title">{data.date.slice(4,)}</h3>
            <h3 className="list-item__title">${data.price}</h3>
            <h3 className="list-item__title">
                {recomand === 1 ? 'Buy' : (recomand === 0 ? 'Hold' : 'Sell')}
            </h3>
        </div>
    )
}

export default Item