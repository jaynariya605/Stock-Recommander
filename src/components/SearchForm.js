import { useState, useContext } from "react";
import data from "../fixtures/data";
import Application from "../context/Application";


export const SearchForm = ({fetchSearchQuery}) => {

    const [search, setSearch] = useState('')
    const [filtered, setFilterd] = useState([])
    const { dispatch } = useContext(Application)

    //Change list to customize socialmedia platform according to user prefernce
    const socialMediaList = ['Facebook']

    const onChangeSearch = (e) => {
        const text = e.target.value.toLowerCase().trim()
        setSearch(text)
        setFilterd(fetchSearchQuery(text))
        if(!text){
            setFilterd([])
        }
    }

    

    const onClickSearch = (e) => {
        const action = {
            type: 'ADD_TITLE',
            title: e.target.value
        }
        dispatch(action)
        setSearch('')
        setFilterd([])
    }

    return (
        <div className="content-container">
            <div className="input-group">
                <div className="input-group__item">
                    <input defaultValue={search} onChange={onChangeSearch} placeholder="Search Stock" className="text-input" type="text"/>
                    { filtered && 
                        <div className="input-dropdown"> 
                            {filtered.map((symbole)=> <button key={symbole} value={symbole} onClick={onClickSearch}>{symbole}</button>)}
                        </div>}
                </div>
                <div className="input-group__item">
                    <select className="select" >
                        {socialMediaList.map((platform)=> <option key={platform} value="date">{platform}</option>)}   
                    </select>
                </div>
            </div>
        </div>
    )
}


//Replace with Elastic search Api to get list of Stock symbols availabe in data set
const fetchSearchQuery = (text) => {
    return Object.keys(data).filter((symbole)=> symbole.includes(text))
}


export default () => {
    return <SearchForm fetchSearchQuery={fetchSearchQuery}/>
}