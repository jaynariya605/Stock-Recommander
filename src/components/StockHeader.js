import {  useContext } from "react";
import Application from "../context/Application";

export const StockHeader= ({state}) => {
  
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title"><span>{state && state.title}</span> </h1>    
            </div>
            
        </div>
    )
}

export default ()=> {
    const { state } = useContext(Application)
    return <StockHeader state={state}/>
}