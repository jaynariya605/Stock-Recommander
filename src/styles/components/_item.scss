
.list-item {
    border: 1px solid darken($off-white, 7%);
    border-top: none;
    color: $dark-grey;
    display: flex;
    flex-direction: column;
    padding: $s-size;
    transition: background .3s ease ;
    &:hover {
        background: $off-white;
    }

    @media(min-width: $desktop-breakpoint){
        align-items: center;
        flex-direction: row;
        justify-content: space-between;
        padding: $m-size;
    }
}
.list-body {
    margin-bottom: $m-size;
    @media(min-width: $desktop-breakpoint){
        margin-bottom: $l-size;
    }
}

.list-item__title {
    margin: 0;
    word-break: break-all;
}

.list-item__subtitle {
    color: $grey;
    font-size: $font-size-small;

}