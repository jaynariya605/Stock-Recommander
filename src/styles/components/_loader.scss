.loader {
	display: flex;
	justify-content: center;
	align-items: center;
	margin-top: $s-size;
}

.loader-circle-2 {
	position: relative;
	width: 40px;
	height: 40px;
	display: inline-block;
}
.loader-circle-2:before,
.loader-circle-2:after {
	content: "";
	display: block;
	position: absolute;
	border-width: 5px;
	border-style: solid;
	border-radius: 50%;
}
.loader-circle-2:before {
	width: 40px;
	height: 40px;
	border-bottom-color: $dark-blue;;
	border-right-color: $dark-blue;;
	border-top-color: transparent;
	border-left-color: transparent;
	animation: loader-circle-2-animation-2 1s linear infinite;
}
.loader-circle-2:after {
	width: 20px;
	height: 20px;
	border-bottom-color: #646464;
	border-right-color: #646464;
	border-top-color: transparent;
	border-left-color: transparent;
	top: 22%;
	left: 22%;
	animation: loader-circle-2-animation 0.85s linear  infinite;
}
@keyframes loader-circle-2-animation {
	0% {
		transform: rotate(0deg);
	}
	100% {
		transform: rotate(-360deg);
	}
}
@keyframes loader-circle-2-animation-2 {
	0% {
		transform: rotate(0deg);
	}
	100% {
		transform: rotate(360deg);
	}
}