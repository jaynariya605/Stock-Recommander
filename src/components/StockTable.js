
import Application from "../context/Application";
import { useContext, useEffect, useState } from "react";
import data from "../fixtures/data";
import Item from "./Item";
import recommandation from "../algorithm/recommandation";


export const StockTable = ( { getData, state }) => {

    
    const [stockdata, setStockData] = useState([])
    const [loading, setLoading] = useState()

    

    useEffect(()=> {

        const fetchData = async () => {
            setLoading(true)
            return new Promise((resolve, reject)=>{
                //settime our for two seccongto see loader
                setTimeout(()=>{
                    resolve(getData(state.title))
               }, 2000)
            })
        }

        if(state){
            setStockData([])
            fetchData().then((data)=>{
                setLoading(false)
                setStockData(data)
            })  
        }
        
    },[state])
    
   
  

    return (
        <div className="content-container">
            <div className="list-header">
                <div className="show-for-mobile">Date</div>
                <div className="show-for-desktop">Price</div>
                <div className="show-for-desktop">suggestion</div>
            </div>
            {loading && (
                <div className="loader">
                    <div class="loader-circle-2"></div>
                </div>
                
            )}

            {stockdata.length !== 0 && stockdata.slice(10).map((item, index)=> {
                return (
                    <Item key={index} recomand={recommandation({data: stockdata, item, index})} data={item}/>
                )
            }) 
        }
        </div>
    )

}

//replace with API function to get data via stock symbole with async call and Promise
const getData = async (symbole) => {
    return data[symbole]
}

export default () => {
    const { state } = useContext(Application)
    return <StockTable getData={getData} state={state} />
}