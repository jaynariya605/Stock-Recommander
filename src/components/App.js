import React, { useReducer } from "react";
import { Header } from "./Header";
import SearchForm from "./SearchForm";
import StockTable from "./StockTable";
import StockHeader from "./StockHeader";
import Application from "../context/Application";
import stockheader from "../reducer/stockheader";

const App = () =>  {
    const [state, dispatch ] = useReducer(stockheader)
    
    return (
        <div>
        <Application.Provider value={ { state, dispatch }} >
            <Header/>
            <StockHeader/>
            <SearchForm/>
            <StockTable/>
        </Application.Provider>
        </div>
    )
}

export default App