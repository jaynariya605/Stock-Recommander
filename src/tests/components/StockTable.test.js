import { StockTable } from "../../components/StockTable";

import { shallow, configure } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';

configure({
    adapter: new Adapter()
})

let wrapper , getData

beforeEach(()=>{
    getData = jest.fn()
    wrapper = shallow(<StockTable getData={getData} state={null}/>)
})

test('should render Stock Table correctly', ()=>{
    expect(wrapper).toMatchSnapshot()
})
