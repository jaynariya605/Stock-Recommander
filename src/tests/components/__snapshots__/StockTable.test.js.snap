// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Stock Table correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Date
    </div>
    <div
      className="show-for-desktop"
    >
      Price
    </div>
    <div
      className="show-for-desktop"
    >
      suggestion
    </div>
  </div>
</div>
`;
